framework/src/wifi_manager/wifi_manager.c

```diff
@@ -17,129 +17,320 @@
  ****************************************************************************/
 
 #include <tinyara/config.h>
-#include <debug.h>
-#include <fcntl.h>
-#include <mqueue.h>
-#include <stdio.h>
-#include <arpa/inet.h>
+#include <pthread.h>
 #include <wifi_manager/wifi_manager.h>
 #include <protocols/dhcpc.h>
 #include <protocols/dhcpd.h>
-#include <slsi_wifi/slsi_wifi_api.h>
-#include <semaphore.h>
+#ifdef CONFIG_ENABLE_IOTIVITY
+#include <mqueue.h>
+#include <fcntl.h>
+#endif
 #include "wifi_utils.h"
-#include "wifi_mutex.h"
-#include "wifi_net.h"
 
-static void *g_dhcp_handle = NULL;
-static dhcp_sta_joined g_new_sta_join = NULL;
+#define NUM_OF_STATE_HANDLER 10 // the number of states
+enum _wifimgr_state {
+	WIFIMGR_NOT_INIT,
+	WIFIMGR_STA_DISCONNECT,
+	WIFIMGR_STA_DISCONNECTING,
+	WIFIMGR_STA_CONNECTING,
+	WIFIMGR_STA_CONNECT,
+	WIFIMGR_SOFTAP, // 5
+	WIFIMGR_SCANNING,
+	WIFIMGR_NONE, // it is used for prev state only
+};
+typedef enum _wifimgr_state _wifimgr_state_e;
+
+enum _wifimgr_evt {
+	EVT_INIT,				// Command to initialize WiFi Manager
+	EVT_DEINIT,				// Command to Deinit WiFi Manager
+	EVT_SET_SOFTAP,			// Command to set SoftAP
+	EVT_SET_STA,			// Command to set STA mode
+	EVT_CONNECT,			// Command to connect to a WiFi AP
+	EVT_DISCONNECT,		// Command to Disconnect from a connected WiFi AP
+	EVT_SCAN,				// Command to perform WiFi Scanning over WLAN channels
+	EVT_GETINFO,			// Command to get WiFi Manager information
+	EVT_STA_CONNECT,		// Event that STA is connected
```

Comment:
to be differentiated from commands, why don't you use past tense, for example, renaming EVT_STA_CONNECT with EVT_STA_CONNECTED?
Another option may be possible.
Just split _wifimgr_evt structure into two: one for commands (or requests) and the other for events.

Reply:
ok I will change